README.md

```diff
@@ -1,178 +1,114 @@
+  A web app providing an editing experience on bot assets. The application has access to any bots made available to it from the `bots` property in `config.json`. See below for more details.
 
-## key points
+  The application also has the ability to start the bot via the BotLauncher. This is a common scenario after one has used the application's dialog editing capabilities and wants to see the current state of the dialogs through conversation with the bot's runtime.
 
-In this one-direction dependency design. A few key points about this kind of architecure that help understand are that
+####launcher
 
+  On command, the launcher will load an bot from `Bots` folder, and start a bot instance and have it ready to serve at a public endpoint. The application will then be able to connect to it during a conversational design session.
 
-* `Bots` holds ground truth.
+  The launcher watches the bot assets (.lu, .lg, .dialog) and reloads the bot runtime when a change in these files are observed.
 
-  Which means everything about the bot, not composer, not bot launcher. 
-  
-  Composer and bot launcher can be viewed as stateless applications, which only have a reference to bots. 
+####bots
+Each bot is defined by a ".bot" file, which includes all the references to the assets this bot would use. **note** this is not the shape of the .bot file currently used in SDK v4 of Bot Framework Emulator v4. The structure of the .bot file is subject to change. 
 
-* `Bot launcher` don't design with `composer` in mind. 
+## data flow
 
-  Bot launcher only depends and focus on `bots`. Nothing to do with composer. Even though composer will be able to run the launcher. 
+The bot's assets are the source of truth for the bot at any given time. The Application will hold dirty bot dialogs that will be lost until written into the bot's assets. The BotLauncher only references the state of the bot's assets and cannot read dirty bot assets held in the application.
 
-* `Composer` knows everything about `bots` and `bot launcher` 
-  
-  Composer needs to know how to start the launcher with property configuration to launcher specific bot. 
+<Needs Image>
 
 ## folder structure
 
-Then let's explore the structure of each component, and dive into some config files to show how this kind of intertion is supported by those files. 
-
-Here is a overview of the folder structure, with hightlighting on some important config files. 
+Here is a overview of a potential folder structure
 
     /- BotLauncher
       /- CSharp
-        /- appsettings.json (has a ref to xx.bot, otherwise which bot to run?)
-    /- Bots
-      /- SampleBot1
-        /- bot1.bot    (bot project file, maybe we can call it .botproject, all files should be referenced here)
+      /- Node
     /- Composer
       /- config.json   (the config to the composer)
-      /- launcher.json (composer use this config to start Bot Launcher)
+    ...somewhere else on disk
+    /-echo-bot
```

Comment:
an example of being able to reference a .bot file somewhere other than `/Bots`

Reply:
the reference to .bot file, is defined in the "bot“ section in config.json, reference to any .bot file from any storage provider